src/app/researcher/study/request/[memberSlug]/study-proposal.tsx

```diff
@@ -5,19 +5,130 @@ import { Button, Group, Stack, Text } from '@mantine/core'
+    const studyProposalForm = useForm<StudyProposalFormValues>({
+        mode: 'uncontrolled',
+        validate: zodResolver(studyProposalFormSchema),
+        initialValues: {
+            title: '',
+            piName: '',
+            irbDocument: null,
+            descriptionDocument: null,
+            agreementDocument: null,
+            codeFiles: [],
+        },
+    })
+
+    const { isPending, mutate: createStudy } = useMutation({
         mutationFn: async (formValues: StudyProposalFormValues) => {
-            return await onCreateStudyAction({ memberId, studyInfo: formValues })
+            // Don't send any actual files to the server action, because they can't handle the file sizes
+            const valuesWithoutFiles = omit(formValues, [
+                'agreementDocument',
+                'descriptionDocument',
+                'irbDocument',
+                'codeFiles',
+            ])
+
+            const valuesWithFilenames = {
+                ...valuesWithoutFiles,
+                descriptionDocPath: formValues.descriptionDocument?.name || '',
+                agreementDocPath: formValues.agreementDocument?.name || '',
+                irbDocPath: formValues.irbDocument?.name || '',
+            }
+
+            const { studyId, studyJobId } = await onCreateStudyAction({
+                memberSlug,
+                studyInfo: valuesWithFilenames,
+            })
+
+            if (formValues.irbDocument) {
+                const path = pathForStudyDocuments({ studyId, memberSlug }, StudyDocumentType.IRB)
+                const upload = await signedUrlForStudyFileUploadAction(path)
+
+                const ok = await uploadFile(formValues.irbDocument, upload)
+                if (!ok) return { studyId, studyJobId }
+            }
+
+            if (formValues.agreementDocument?.name) {
+                const path = pathForStudyDocuments({ studyId, memberSlug }, StudyDocumentType.AGREEMENT)
+                const upload = await signedUrlForStudyFileUploadAction(path)
+
+                const ok = await uploadFile(formValues.agreementDocument, upload)
+                if (!ok) return { studyId, studyJobId }
+            }
+
+            if (formValues.descriptionDocument?.name) {
+                const path = pathForStudyDocuments({ studyId, memberSlug }, StudyDocumentType.DESCRIPTION)
+                const upload = await signedUrlForStudyFileUploadAction(path)
+
+                const ok = await uploadFile(formValues.descriptionDocument, upload)
+                if (!ok) return { studyId, studyJobId }
+            }
+
+            const studyCodeUpload = await signedUrlForCodeUploadAction({
+                memberSlug,
+                studyId,
+                studyJobId,
+            })
+
+            const ok = await uploadCodeFiles(formValues.codeFiles, studyCodeUpload, studyJobId)
+            if (!ok) return { studyId, studyJobId }
+
+            return { studyId, studyJobId }
```

Comment:
is it possible to shorten all this, having this much code in the body of a component gives me shakes.  At first I was going to suggest exacting it but instead I think we can make a `uploadFiles` function that accepts an array of Files & paths and cut all this down to:
```suggestion
        uploadFiles(signedUrlFromStudySave, [
            { path: pathForStudyDocuments({ studyId, memberSlug }, StudyDocumentType.IRB),  file: formValues.irbDocument },
            {  const path = pathForStudyDocuments({ studyId, memberSlug }, StudyDocumentType.AGREEMENT), file:  formValues.agreementDocument }
            ...etc
])            
```

Reply:
Ooh, a further thought:  we can have the save study action generate the exact url for each file and return something like:
```
{
 irb:  { uploadUrl: <signed url>, deleteUrl: <signed url> }
 agreement:  { uploadUrl: <signed url>, deleteUrl: <signed url>  }
 codeFiles: [{  uploadUrl: <signed url>, deleteUrl: <signed url> }]
}
```

Reply:
a FINAL thought: maybe we don't need delete urls returned by study save.  if uploads fail, we'll need to call a 2nd `deleteStudyAction` and that could return the delete urls